### PR TITLE
Address Banner and MREC ad display issue.

### PR DESCRIPTION
**Pull request by ManojBudumuru-vungle**
This commit will update the show ad for Banner/Mrec to be called after the ad finishes downloading the assets.

IOS-5559
IOS-5560

### CODE REVIEW

**Review by stanley-vungle — APPROVED** [review]
LGTM

**Review by ashinagawa — COMMENTED** [review]
I could verify that IOS-5559 has been fixed.  Banner and MREC are displayed after fresh install.

But, I still see the IOS-5560 issue.
After displaying MREC first time, and go back to main screen, then go to "MRECs> AutoLayout MRECs" and no MREC displayed this time.

**Review by YueVungle — APPROVED** [review]
LGTM 
@ashinagawa  @ManojBudumuru-vungle  I displayed the Vungle MREC Ad after the first time MREC Ad played. But for MREC Ads, I got Applovin ads most of the time. Thnanks :)

**Comment by YueVungle** [comment]
Test again. For MREC Ads, still got Applovin ads most of the time. Thanks :)

**Comment by stanley-vungle** [comment]
Are you using a physical device? If I use the Simulator, I only get AppLovin ads, but on a physical device, I can get vungle ads.

File: Vungle/VungleAdapter/ALVungleMediationAdapter.m [hunk 1]
```diff
@@ -140,7 +141,14 @@ - (void)destroy
     if ( self.adView )
     {
         // Note: Not calling this for now because it clears pre-loaded/cached ad view ads as well.
-        // [[VungleSDK sharedSDK] finishedDisplayingAd];
+        if (self.bidResponse) {
```
Comment (stanley-vungle) [inline Vungle/VungleAdapter/ALVungleMediationAdapter.m]: Need to add spaces and move the brace to the other line to follow MAX standards

Comment (YueVungle) [reply Vungle/VungleAdapter/ALVungleMediationAdapter.m]: Line 1122：
```
MANativeAd *maxNativeAd = [[MAVungleNativeAd alloc] initWithParentAdapter: self.parentAdapter builderBlock:^(MANativeAdBuilder *builder) {
            builder.title = nativeAd.title;
            builder.body = nativeAd.bodyText;
```
Might need to move the brace to the other line to keep the format consistent. Thanks. :)

**Comment by YueVungle** [comment]
> Are you using a physical device? If I use the Simulator, I only get AppLovin ads, but on a physical device, I can get vungle ads.

@stanley-vungle Thanks a lot for the reminder, changed to device. :)

File: Vungle/VungleAdapter/ALVungleMediationAdapter.m [hunk 2]
```diff
@@ -444,8 +454,17 @@ - (void)loadAdViewAdForParameters:(id<MAAdapterResponseParameters>)parameters ad
 {
     NSString *bidResponse = parameters.bidResponse;
     BOOL isBiddingAd = [bidResponse al_isValidString];
+    if ( isBiddingAd && bidResponse )
+    {
+        if ( !self.adViewbidResponseDict )
+        {
+            self.adViewbidResponseDict = [[NSMutableDictionary alloc] init];
+        }
+        self.adViewbidResponseDict[self.placementIdentifier] = bidResponse;
```
Comment (YueVungle) [inline Vungle/VungleAdapter/ALVungleMediationAdapter.m]: @ManojBudumuru-vungle Tested on the device, for bidding banner, failed to load ad with the error:
```
2022-11-11 17:17:18.002401+0800 AppLovin MAX Demo App - ObjC[1526:55660] [AppLovinSdk] ERROR [ALMediationAdapterWrapper] Failed start loading [ALMediatedAdViewAd thirdPartyAdPlacementIdentifier=IAB_BANNER_1-6722317, adUnitIdentifier=55f1fdf557bea516, format=BANNER, networkName=Vungle] : *** -[__NSDictionaryM setObject:forKeyedSubscript:]: key cannot be nil
(lldb) po error
[MAError: code=-5001, message=MAX returned eligible ads from mediated networks, but all ads failed to load. Inspect -[MAError waterfall] for more info., mediatedNetworkErrorCode=-1, mediatedNetworkErrorMessage=]
```
Trough debugging, found that the `self.placementIdentifier` is `nil` when `self.adViewbidResponseDict[self.placementIdentifier] = bidResponse;`, we need to put `self.placementIdentifier = parameters.thirdPartyAdPlacementIdentifier;` before `self.adViewbidResponseDict[self.placementIdentifier] = bidResponse;`

After put `self.placementIdentifier = parameters.thirdPartyAdPlacementIdentifier;` before `self.adViewbidResponseDict[self.placementIdentifier] = bidResponse;` We can receive Bidding Banner ads:
ad

Thanks a lot. :)

Comment (ManojBudumuru-vungle) [reply Vungle/VungleAdapter/ALVungleMediationAdapter.m]: Good catch. i forgot to remove this. we dont need it.

**Pull request merged**